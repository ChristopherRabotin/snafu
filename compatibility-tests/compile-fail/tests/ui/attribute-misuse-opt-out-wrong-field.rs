mod source {
    use snafu::prelude::*;

    #[derive(Debug, Snafu)]
    enum EnumError {
        AVariant {
            #[snafu(source(false))]
            not_source: u8,
        },
    }
}

mod backtrace {
    use snafu::prelude::*;

    #[derive(Debug, Snafu)]
    enum EnumError {
        AVariant {
            #[snafu(backtrace(false))]
            not_backtrace: u8,
        },
    }
}

fn main() {}
