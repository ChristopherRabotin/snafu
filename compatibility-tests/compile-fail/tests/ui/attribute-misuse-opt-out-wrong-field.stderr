error: `source(false)` attribute is only valid on a field named "source", not on other fields
 --> $DIR/attribute-misuse-opt-out-wrong-field.rs:7:21
  |
7 |             #[snafu(source(false))]
  |                     ^^^^^^^^^^^^^

error: `backtrace(false)` attribute is only valid on a field named "backtrace", not on other fields
  --> $DIR/attribute-misuse-opt-out-wrong-field.rs:19:21
   |
19 |             #[snafu(backtrace(false))]
   |                     ^^^^^^^^^^^^^^^^

error: `implicit(false)` attribute is only valid on a field named "location", not on other fields
  --> $DIR/attribute-misuse-opt-out-wrong-field.rs:31:21
   |
31 |             #[snafu(implicit(false))]
   |                     ^^^^^^^^^^^^^^^
